pkg/services/auth/auth_token.go

```diff
@@ -404,6 +414,36 @@ func (s *UserAuthTokenService) rotatedAfterParam() int64 {
 	return getTime().Add(-s.Cfg.LoginMaxInactiveLifetime).Unix()
 }
 
+func (s *UserAuthTokenService) revokeOldUserTokens(ctx context.Context, userId int64, maxConcurrentSessions int) error {
+	return s.SQLStore.WithDbSession(ctx, func(dbSession *sqlstore.DBSession) error {
+		sql := `
+		DELETE
+		FROM user_auth_token
+		WHERE id NOT IN
+			(
+			  SELECT id
+			  FROM user_auth_token
+			  WHERE user_id = ?
+			  ORDER BY seen_at DESC
+			  LIMIT ?
+			)
+		AND user_id = ?
+		AND seen_at != 0;`
```

Comment:
Not sure about the logic for `seen_at`. Maybe moving this above inserting a new token would simplify things? Doing that could do a query to receive all auth tokens for user and then iterate and find the oldest (based on created_at or rotated_at) above `maxConcurrentSessions-1`  token and finally `delete from user_auth_token where id in(<list of id's>)`. 

Doing this you could potentially move all of this logic to enterprise by wrapping and overriding the `UserTokenService` in similar manner as done with the [data source cache service](https://github.com/grafana/grafana/blob/master/pkg/services/datasources/cache.go) using [registry.RegisterOverride](https://github.com/grafana/grafana/blob/master/pkg/registry/registry.go#L45) (in enterprise). @joanlopez @xlson thoughts?

Reply:
Yup! I also thought about it, but I did it before just to avoid removing an existing session with the possibility of having issues while creating the new one. But this might be something insignificant for this case. So, if it helps us to simplify the code, I agree we can do it the other way round.

Reply:
> Doing this you could potentially move all of this logic to enterprise by wrapping and overriding the UserTokenService ª in similar manner as done with the data source cache service using registry.RegisterOverride (in enterprise). @joanlopez @xlson thoughts?

This approach produces `inject` errores because the `models.UserTokenService` interface is injected into some other services and once both implementations are present it does not know which one should be injected.

Any idea? 🤔  cc/ @marefr @xlson @sakjur 

Thank you so much!!! 🙏